### PR TITLE
Allow open ranges for chapters

### PR DESCRIPTION
**Pull request by Lesmiscore**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

This will allow `--download-sections` and its relatives to allow using range with the other end missing (e.g. `*2-`, `*-60`)

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [x] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/__init__.py [hunk 1]
```diff
@@ -329,8 +329,13 @@ def parse_chapters(name, value):
...
                 for range in regex[1:].split(','):
-                    dur = tuple(map(parse_duration, range.strip().split('-')))
-                    if len(dur) == 2 and all(t is not None for t in dur):
+                    range = range.strip()
+                    dur = tuple(map(parse_duration, range.split('-')))
+                    if len(dur) == 2 and range != '-' and (
+                            # start-end
+                            all(t is not None for t in dur)
+                            # start- or -end
+                            or any(x == '-' for x in (range[0], range[-1]))):
```
Comment (pukkandan) [inline yt_dlp/__init__.py:338]: Since it got this complicated, better to use a regex imo

Comment (Lesmiscore) [reply yt_dlp/__init__.py:338]: you meant `(\d+)?-(\d+)?` (without r')

File: yt_dlp/YoutubeDL.py [hunk 2]
```diff
@@ -2720,8 +2721,8 @@ def to_screen(*msg):
                 offset, duration = info_dict.get('section_start') or 0, info_dict.get('duration') or float('inf')
                 if chapter or offset:
                     new_info.update({
-                        'section_start': offset + chapter.get('start_time', 0),
-                        'section_end': offset + min(chapter.get('end_time', duration), duration),
+                        'section_start': offset + chapter.get('start_time', 0) if chapter.get('start_time') is not None else None,
+                        'section_end': offset + min(chapter.get('end_time', duration), duration) if chapter.get('end_time') is not None else None,
```
Comment (pukkandan) [inline yt_dlp/YoutubeDL.py:2725]: Wouldn't this be much simpler if we parse empty range to `0` and `float('inf')`? That would also simplify the above printing

Comment (Lesmiscore) [reply yt_dlp/YoutubeDL.py:2725]: That way it complicates FFmpegFD (especially the duration calculation for `-t`)

**Comment by pukkandan** [comment]
Can merge if you have no issues with my changes

**Pull request merged**